### ISSUE TITLE
4.0.0-beta 3 does not show icon in torrent list for single file torrents that have existing icons on Windows 11

### ISSUE THREAD

**Issue by Pentaphon** [transmission/transmission#4566]
### What is the issue?

![icon](https://user-images.githubusercontent.com/49374058/211233855-092af445-2035-4e13-883d-f0afba957cee.png)

I thought it was just .iso files that had this issue but then it seems all video files have and single file icons just don't have an icon show up in the torrent list. I didn't notice before because I usually download folders. 

![big](https://user-images.githubusercontent.com/49374058/211234383-b4b8db99-691d-446f-aa62-29cb20d883ed.png)

The icons do show up fine in the file viewer in the properties window, though. 

![file](https://user-images.githubusercontent.com/49374058/211234561-abeee245-bd0f-4a5f-bf3b-67dcb07faaf0.png)

### Which application of Transmission?

Qt app on Windows

### Which version of Transmission?

4.0.0-beta 3

**Comment by ckerr** [comment]
This is currently trying to get an icon for the mime type, e.g. `video/quicktime` and `application/octet-stream`, and it looks like maybe QMimeType is failing to get icons for that. Maybe we should fall back to `video` and `application`?

**Comment by ckerr** [comment]
Oh hmm we already are using a fallback. So why is this failing, I wonder...

https://github.com/transmission/transmission/blob/main/qt/IconCache.cc#L83

**Comment by Pentaphon** [comment]
> Oh hmm we already are using a fallback. So why is this failing, I wonder...

Perhaps @mikedld can shed some light on this?

**Comment by mikedld** [comment]
https://doc.qt.io/qt-6/qicon.html#fromTheme

> **Note:** By default, only X11 will support themed icons. In order to use themed icons on Mac and Windows, you will have to bundle a compliant theme in one of your [themeSearchPaths](https://doc.qt.io/qt-6/qicon.html#themeSearchPaths)() and set the appropriate [themeName](https://doc.qt.io/qt-6/qicon.html#themeName)().

There's a Windows-specific `IconCache::addAssociatedFileIcon()` but it's not being used for icons in main window since #1464.

**Comment by Pentaphon** [comment]
> There is no API on Windows (and Mac?) to get an icon based on a MIME type that I know of (feel free to do your own search), and MIME types is what libtransmission returns. Windows does have a way of getting icons based on file names/extensions, and it's what was used before.

If it was used before, why can't it be used again? I'm confused as to why it isn't used in 4.x.x.

**Comment by bananakid** [comment]
> @bananakid can you point out this regression?

@Pentaphon would you kindly elaborate what data I should gather? I can double-check any version once again, however IIRC it's everything after 3.0.0 that got broken icon-wise.



> If using non-system icons is acceptable, the solution is to bundle [much] more icons from a chosen icon pack (which is currently Faenza, https://github.com/transmission/transmission/tree/main/qt/icons/Faenza) so that `QIcon::fromTheme()` can find them.

@mikedld thank you for continuous support of Win32 build! I'm not sure bundling Faenza icons is the best way since they don't work very well with Fluent design code used by recent Windows versions. This can be a temporary workaround, however an option (or Qt CLI switch) to disable icons altogether looks like a better alternative meanwhile. Such an option (maybe enabled by default until it's sorted) may provide less frustration to new users who are not familiar with this. I also have noticed the Windows default folder icon is visible for "folder" type of torrents, but it's behind generic blank icon:

![Transmission 4.0.3](https://github.com/transmission/transmission/assets/17095595/a78cfe1e-6b92-487a-bf66-f7bddd05d59f)

**Comment by Pentaphon** [comment]
> would you kindly elaborate what data I should gather? I can double-check any version once again, however IIRC it's everything after 3.0.0 that got broken icon-wise.

@bananakid sorry, I thought you meant you knew what exact code to make icons work in torrent lists got "broken" in 4.0.0 so that you could shine some light on solving this issue. Don't worry about it. I was just hoping you could speed up this fix because right now, 4.0.3 looks very unfinished on Windows with those blank page icons.

**Comment by bananakid** [comment]
@Pentaphon I agree, thank you!

**Comment by Pentaphon** [comment]
> I noticed the preview screenshot at the official website shows correct icons:

Probably a screenshot of 3.0 since the Qt UI hasn't changed...yet.

**Comment by pinkmercedes** [comment]
Hi,

Transmission 4.0.6 38c164933e (qt5/qt6).
Still no icons.
Checked on 10 (x64), 11 (x64), 2012 R2 (x64).
This just ruins user experience on Windows.
Please, fix that.

Thank you!

**Comment by Pentaphon** [comment]
> This just ruins user experience on Windows.

I don't think it ruins the user experience but it certainly makes the Windows Qt client feel unfinished. This is a bug that should have been gone in the betas yet it is still here in what is likely the last 4.0.x version. I hope the admins do something to make this bug more visible like stickying it to the top of the issue page so we get more eyes on it.

**Comment by pinkmercedes** [comment]
Stable Transmission 4.1.0 (272401184f).
No icons.

**Comment by ckerr** [comment]
Could someone on Windows give #8443 a try?

Artifact download URL: https://github.com/transmission/transmission/actions/runs/21892780930/artifacts/5460197373

**Comment by Pentaphon** [comment]
Seeing the same thing as bananakid.

**Comment by mikedld** [comment]
I've already confirmed it working on a limited test set but if you want to help check on a wider set of torrents, grab the binaries built off of `4.1.x` branch with this fix: https://github.com/transmission/transmission/actions/runs/21927257735. Qt client built off of `main` seems to be broken at the moment.

**Comment by Pentaphon** [comment]
> grab the binaries built off of `4.1.x` branch with this fix:

Confirmed working. 

<img width="600" height="215" alt="Image" src="https://github.com/user-attachments/assets/326b4d36-36aa-404d-a9b7-316728445eb4" />

**Comment by ckerr** [comment]
Fixed by #8456, which will be in 4.1.1
